tests/sympc/tensor/mpc_tensor_test.py

```diff
@@ -472,3 +472,25 @@ def test_pow(get_clients, power) -> None:
 
     with pytest.raises(RuntimeError):
         power = x ** -2
+
+
+def test_backward(get_clients):
+    clients = get_clients(4)
+    session = Session(parties=clients)
+    session.autograd_active = True
+    SessionManager.setup_mpc(session)
+
+    x_secret = torch.tensor([[0.125, -1.25], [-4.25, 4], [-3, 3]], requires_grad=True)
+    y_secret = torch.tensor([[4.5, -2.5], [5, 2.25], [-3, 3]], requires_grad=True)
+    x = MPCTensor(secret=x_secret, session=session, requires_grad=True)
+    y = MPCTensor(secret=y_secret, session=session, requires_grad=True)
```

Comment:
Q: If you set `requires_grad=False`, does the `.backward()` not compute anything? 

Maybe: If so, we could have a test to check that...

Reply:
no :))
I was not able to write the test (and was  debugging if it was a issue or I am missing something)

Reply:
basically, if we don't set this to true, `Requires Grad` will be False and `GradFunc` will not be set. So it like calling backward on a normal tensor that has `requires_grad = False.`
![Screenshot 2021-06-09 at 10 51 00 PM](https://user-images.githubusercontent.com/28955148/121400399-2b50bc00-c975-11eb-9c4f-1300f0720794.png)

Reply:
I think @Syzygianinfern0 wanted to validate that we do not have the gradient in that case.
Also, a test case when only one of the parameters has `requires_grad might be good.

x = MPC1 (without `requires grad`)
y = MPC2 (without `requires_grad`)

x - y (check if this one has `requires_grad` and check if `grad` is `None`